task_sdk/tests/execution_time/test_task_runner.py

```diff
@@ -26,13 +26,59 @@
 import pytest
 from uuid6 import uuid7
 
+from airflow.exceptions import AirflowSkipException
 from airflow.sdk import DAG, BaseOperator
 from airflow.sdk.api.datamodels._generated import TaskInstance, TerminalTIState
 from airflow.sdk.execution_time.comms import DeferTask, SetRenderedFields, StartupDetails, TaskState
-from airflow.sdk.execution_time.task_runner import CommsDecoder, parse, run, startup
+from airflow.sdk.execution_time.task_runner import CommsDecoder, RuntimeTaskInstance, parse, run, startup
 from airflow.utils import timezone
 
 
+def get_inline_dag(dag_id: str, task: BaseOperator) -> DAG:
+    """Creates an inline dag and returns it based on dag_id and task."""
+    dag = DAG(dag_id=dag_id, start_date=timezone.datetime(2024, 12, 3))
+    task.dag = dag
+
+    return dag
+
+
+@pytest.fixture
+def mocked_parse(spy_agency):
+    """
+    Fixture to set up an inline DAG and use it in a stubbed `parse` function. Use this fixture if you
+    want to isolate and test `parse` or `run` logic without having to define a DAG file.
+
+    This fixture returns a helper function `set_dag` that:
+    1. Creates an in line DAG with the given `dag_id` and `task` (limited to one task)
+    2. Constructs a `RuntimeTaskInstance` based on the provided `StartupDetails` and task.
+    3. Stubs the `parse` function using `spy_agency`, to return the mocked `RuntimeTaskInstance`.
+
+    Example usage:
+    After adding the fixture in your test function signature, you can use it like this:
+
+    `mocked_parse(StartupDetails(
+        ti=TaskInstance(id=uuid7(), task_id="hello", dag_id="super_basic_run", run_id="c", try_number=1),
+        file="",
+        requests_fd=0,
+    ), "example_dag_id", CustomOperator(task_id="hello"))`
+    """
```

Comment:
```suggestion
    After adding the fixture in your test function signature, you can use it like this:

	    mocked_parse(StartupDetails(
	        ti=TaskInstance(id=uuid7(), task_id="hello", dag_id="super_basic_run", run_id="c", try_number=1),
	        file="",
	        requests_fd=0,
	    ), "example_dag_id", CustomOperator(task_id="hello"))
    """
```

Reply:
FYI: The double trailing colon `::` is one/the easiest way of telling Sphinx that the following block is code, not prose.

Reply:
Yeah i checked the other example and realised. Thanks!